src/components/Header/Header.tsx

```diff
@@ -0,0 +1,107 @@
+import React, {
+  FC, useCallback, useMemo, useState,
+} from 'react';
+import cn from 'classnames';
+import { Errors } from '../../types/Errors';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[],
+  handleError: (error: Errors) => void,
+  addTodo: (inputValue: string) =>void,
+  updateTodo: (updatedTodo: Todo) =>void,
+  setLoadingTodoId: (loadingTodoId: number[]) => void;
+  setErrorMessage: (error: Errors) => void,
+};
+export const Header: FC<Props> = (props) => {
+  const {
+    todos,
+    handleError,
+    addTodo,
+    updateTodo,
+    setLoadingTodoId,
+    setErrorMessage,
+  } = props;
+
+  const [inputValue, setInputValue] = useState('');
+  const [isDisabledInput, setIsDisabledInput] = useState(false);
+
+  const handleFocusOnInput = useCallback((input: HTMLInputElement) => {
+    input?.focus();
+  }, []);
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!inputValue.trim()) {
+      handleError(Errors.EmptyTitle);
+
+      return;
+    }
+
+    setIsDisabledInput(true);
+
+    try {
+      addTodo(inputValue);
```

Comment:
```suggestion
      await addTodo(inputValue);
```
You should wait for the response from the API before clear input

Tip: to have ability to wait for the response from the `addTodo` function it should return the promise.
```typescript
const addTodo = (inputValue: string) => {
    // your code

    return todosService.createTodo(data) // return your api call
      .then().catch().finally() // your logic to handle response from create todo API call
```